### PR TITLE
[Doc][KubeRay] Add a doc for RayService high-availability

### PR DESCRIPTION
**Pull request by kevin85421**
## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/kubernetes/user-guides/rayservice-high-availability.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+(kuberay-rayservice-ha)=
+# RayService high-availability
```
Comment (architkulkarni) [inline doc/source/cluster/kubernetes/user-guides/rayservice-high-availability.md:2]:
```suggestion
# RayService high availability
```

**Review by architkulkarni — APPROVED** [review]
Looks good to me! Just a few nits

File: doc/source/cluster/kubernetes/getting-started/rayservice-quick-start.md [hunk 2]
```diff
@@ -21,7 +21,7 @@ A RayService manages these components:
 * **Kubernetes-native support for Ray clusters and Ray Serve applications:** After using a Kubernetes config to define a Ray cluster and its Ray Serve applications, you can use `kubectl` to create the cluster and its applications.
 * **In-place updating for Ray Serve applications:** See [RayService](kuberay-rayservice) for more details.
 * **Zero downtime upgrading for Ray clusters:** See [RayService](kuberay-rayservice) for more details.
-* **High-availabilable services:** See [RayService](kuberay-rayservice) for more details.
+* **High-availabilable services:** See [RayService high-availability](kuberay-rayservice-ha) for more details.
```
Comment (kevin85421) [inline doc/source/cluster/kubernetes/getting-started/rayservice-quick-start.md:24]:
```suggestion
* **High-availabilable services:** See [RayService high availability](kuberay-rayservice-ha) for more details.
```

File: doc/source/cluster/kubernetes/user-guides/rayservice.md [hunk 3]
```diff
@@ -294,6 +294,7 @@ kubectl delete pod curl
 
 ## Next steps
 
+* See [RayService high-availability](kuberay-rayservice-ha) for more details on RayService HA.
```
Comment (kevin85421) [inline doc/source/cluster/kubernetes/user-guides/rayservice.md:297]:
```suggestion
* See [RayService high availability](kuberay-rayservice-ha) for more details on RayService HA.
```

File: doc/source/cluster/kubernetes/user-guides/rayservice-high-availability.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+### Step 7: Delete the Ray head Pod
+
+```sh
+# Step 7.1: Delete the Ray head Pod.
+export HEAD_POD=$(kubectl get pods --selector=ray.io/node-type=head -o custom-columns=POD:metadata.name --no-headers)
+kubectl delete pod $HEAD_POD
+```
+
+In this example, `query.py` ensures there's at most one in-flight request at any given time.
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/rayservice-high-availability.md:95]:
```suggestion
In this example, `query.py` ensures that at most one request is in-flight at any given time.
```

File: doc/source/cluster/kubernetes/user-guides/rayservice-high-availability.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+Furthermore, there are no Ray Serve replicas on the Ray head Pod.
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/rayservice-high-availability.md:96]:
```suggestion
Furthermore, the Ray head Pod has doesn't have any Ray Serve replicas.
```

File: doc/source/cluster/kubernetes/user-guides/rayservice-high-availability.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+The request may only fail when a request is in the HTTPProxyActor on the Ray head Pod.
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/rayservice-high-availability.md:97]:
```suggestion
Requests may fail only when a request is in the HTTPProxyActor on the Ray head Pod.
```

File: doc/source/cluster/kubernetes/user-guides/rayservice-high-availability.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+Therefore, it's highly probable that no failures occur during the deletion and recovery of the Ray head Pod.
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/rayservice-high-availability.md:98]:
```suggestion
Therefore, failures are highly unlikely to occur during the deletion and recovery of the Ray head Pod.
```

File: doc/source/cluster/kubernetes/user-guides/rayservice-high-availability.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+Users can implement their own retry logic in their Ray scripts to handle the failures.
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/rayservice-high-availability.md:99]:
```suggestion
You can implement retry logic in Ray scripts to handle the failures.
```

File: doc/source/cluster/kubernetes/user-guides/rayservice-high-availability.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
+(kuberay-rayservice-ha)=
+# RayService high availability
+
+[RayService](kuberay-rayservice) provides high availability so that the service can still serve requests when the Ray head Pod fails.
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/rayservice-high-availability.md:4]:
```suggestion
[RayService](kuberay-rayservice) provides high availability (HA) for services to continue serving requests when the Ray head Pod fails.
```

**Review by angelinalg — APPROVED** [review]
Thanks for documenting this! I have some nits. Please let me know if you don't understand any of them. I'm happy to explain or improve them.

File: doc/source/cluster/kubernetes/user-guides/rayservice-high-availability.md [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+Follow [this document](kuberay-operator-deploy) to install the latest stable KubeRay operator by Helm repository.
+
+### Step 3: Install a RayService with GCS fault tolerance enabled
+
+```sh
+curl -LO https://raw.githubusercontent.com/ray-project/kuberay/master/ray-operator/config/samples/ray-service.high-availability.yaml
+kubectl apply -f ray-service.high-availability.yaml
+```
+
+The [ray-service.high-availability.yaml](https://raw.githubusercontent.com/ray-project/kuberay/v1.0.0/ray-operator/config/samples/ray-service.high-availability.yaml) file has several Kubernetes objects:
```
[the remark this answers is not in the document]
Comment (kevin85421) [reply doc/source/cluster/kubernetes/user-guides/rayservice-high-availability.md]: Good point. I decide to point to the master branch before I release v1.0.0.
https://github.com/ray-project/ray/pull/40672/commits/d456ee2c3f1acc21bb6fecc7be0a65b798933452

**Comment by architkulkarni** [comment]
Windows core and serve tests unrelated

**Pull request merged**